controller.go

```diff
@@ -55,9 +55,25 @@ func (c *controller) Run(ctx context.Context) error {
 			worker.Run(workersCtx)
 		}()
 	}
+	errChan := make(chan error, 1)
 	streamCtx, cancelStream := context.WithCancel(ctx)
 	// Run streams subscribers
 	c.Info("Wait to become leader")
+
+	defer func() {
+		c.Info("stopping controller...")
+		c.Info("stopping streams...")
+		cancelStream()
+		c.streamWaitGroup.Wait()
+		c.Info("stopped streams...")
+		c.Info("stopping workers...")
+		cancelWorkers()
+		c.workerWaitGroup.Wait()
+		c.Info("stopped workers...")
+		c.Info("stopped controller...")
+		close(errChan)
```

Comment:
Is there a specific motiviation for moving this into defer?
I find defer makes the code harder to read so I prefer to avoid it when possible (an `err` var might just make things work)

Reply:
The idea was to be able to return directly the error instead of storing and returning the error later (my previous version not commited).